app/src/main/java/org/wikipedia/onboarding/InitialOnboardingFragment.kt

```diff
@@ -0,0 +1,136 @@
+            advancePage()
+        } else {
+            super.onActivityResult(requestCode, resultCode, data)
+        }
+    }
+
+    override fun onSwitchChange(view: OnboardingPageView, checked: Boolean) {
+        if (OnboardingPage.of(view.tag as Int) == OnboardingPage.PAGE_USAGE_DATA) {
+            Prefs.setEventLoggingEnabled(checked)
+        }
+    }
+
+    override fun onLinkClick(view: OnboardingPageView, url: String) {
+        when (url) {
+            "#login" -> startActivityForResult(LoginActivity
+                    .newIntent(requireContext(), LoginFunnel.SOURCE_ONBOARDING),
+                    Constants.ACTIVITY_REQUEST_LOGIN)
+            "#privacy" -> FeedbackUtil.showPrivacyPolicy(requireContext())
+            "#about" -> FeedbackUtil.showAboutWikipedia(requireContext())
+            "#offline" -> FeedbackUtil.showOfflineReadingAndData(requireContext())
+            else -> UriUtil.handleExternalLink(requireActivity(), Uri.parse(url))
+        }
+    }
+
+    override fun onListActionButtonClicked(view: OnboardingPageView) {
+        onboardingPageView = view
+        requireContext().startActivity(WikipediaLanguagesActivity.newIntent(requireContext(), Constants.InvokeSource.ONBOARDING_DIALOG))
+    }
+
+    override fun onResume() {
+        super.onResume()
+        onboardingPageView?.refreshLanguageList()
+    }
+
+    private class OnboardingPagerAdapter constructor(fragment: Fragment) : FragmentStateAdapter(fragment) {
+        override fun createFragment(position: Int): Fragment {
+            return ItemFragment.newInstance(position)
+        }
+
+        override fun getItemCount(): Int {
+            return OnboardingPage.size()
+        }
+    }
+
+    class ItemFragment : Fragment() {
+        override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
+            super.onCreateView(inflater, container, savedInstanceState)
+            val position = requireArguments().getInt("position", 0)
+            val view = inflater.inflate(OnboardingPage.of(position).layout, container, false) as OnboardingPageView
+            if (OnboardingPage.PAGE_USAGE_DATA.code() == position) {
+                view.setSwitchChecked(Prefs.isEventLoggingEnabled())
+            }
+            view.tag = position
+            view.callback = callback
+            return view
+        }
+
+        private val callback
+            get() = FragmentUtil.getCallback(this, OnboardingPageView.Callback::class.java)
+
+        companion object {
+            fun newInstance(position: Int): ItemFragment {
+                val instance = ItemFragment()
+                val args = Bundle()
+                args.putInt("position", position)
+                instance.arguments = args
+                return instance
+            }
+        }
+    }
+
+    internal enum class OnboardingPage(@LayoutRes val layout: Int) : EnumCode {
```

Comment:
Since the items are not really unused, would it be better if we add `@Suppress("unused")` to reduce confusion?